File: src/dp/_674_find_length_of_lcis.rs
struct Solution;

impl Solution {
    pub fn find_length_of_lcis(nums: Vec<i32>) -> i32 {
        if nums.len() == 0 {
            return 0;
        }
        let mut ret = 1_i32;
        let mut count = 1_i32;
        for i in 0..nums.len() - 1 {
            if nums[i + 1] > nums[i] {
                count += 1;
            } else {
                count = 1;
            }
            if count > ret {
                ret = count;
            }
        }
        ret
    }
}

File: leetcode-go/674_findLengthOfLCIS.go
package leetcode_go

func findLengthOfLCIS(nums []int) int {
	dp := make([]int, len(nums))
	for i := range dp {
		dp[i] = 1
	}
	ret := 1
	for i := 0; i < len(nums)-1; i++ {
		if nums[i+1] > nums[i] {
			dp[i+1] = dp[i] + 1
			if dp[i+1] > ret {
				ret = dp[i+1]
			}
		}
	}
	return ret
}
